iq_learn/dataset/expert_dataset.py

```diff
@@ -88,7 +88,10 @@ def __init__(self,
         # apply permutation to cond
         # print("perm:",perm)
         if cond_type!="none":
-            self.conds = [conds["emb"][i] for i in perm]
+            if type(conds["emb"][0])==torch.Tensor:
+                self.conds = [conds["emb"][i].detach().cpu().numpy() for i in perm]
```

Comment:
does it affect the gradient back propagation?

Reply:
No. The probabilistic encoder generate cond and dist_params only using the traj index from buffer instead of directly the cond from buffer.